contentcuration/contentcuration/static/js/edit_channel/vuexModules/asyncTask.js

```diff
@@ -0,0 +1,86 @@
+const DEFAULT_CHECK_INTERVAL = 5000;
+const TASKS_RUNNING_CHECK_INTERVAL = 1000;
+
+let timerID = null;
+
+const asyncTasksModule = {
+  state: {
+    asyncTasks: [],
+    runningTasks: [],
+    finishedTasks: [],
+  },
+  getters: {
+    asyncTasks(state) {
+      return state.asyncTasks;
+    },
+    runningTasks(state) {
+      return state.runningTasks;
+    },
+  },
+  actions: {
+    startTask(store, newTask) {
+      if (timerID) {
+        clearTimeout(timerID);
+      }
+      let tasks = store.getters.asyncTasks;
+      tasks.push(newTask);
+      store.dispatch('updateTaskList');
+    },
+
+    updateTaskList(store) {
+      $.ajax({
+        method: 'GET',
+        url: '/api/task',
+        dataType: 'json',
+        success: function(data) {
```

Comment:
Is there a risk that there are a lot of tasks running at the same time and this will cause a lot of slowdown? Guessing the permissions must filter out tasks based on the user, but just wanted to check here